File: src/main/frontend/helper/MySignalProviderService.ts
import { NumberSignalChannel } from "@vaadin/hilla-react-signals";
import client_1 from "Frontend/generated/connect-client.default.js";

export class MySignalProviderService {
  static sharedValue() {
    const signalChannel =
      new NumberSignalChannel('MySignalProviderService.sharedValue', client_1);
    return signalChannel.signal;
  }

  static counter() {
    const signalChannel =
      new NumberSignalChannel('MySignalProviderService.counter', client_1);
    return signalChannel.signal;
  }
}

File: src/main/java/com/github/taefi/services/MySignalProviderService.java
package com.github.taefi.services;

import com.vaadin.hilla.signals.NumberSignal;
import com.vaadin.flow.server.auth.AnonymousAllowed;
import com.vaadin.hilla.BrowserCallable;

@AnonymousAllowed
@BrowserCallable
public class MySignalProviderService {

    private final NumberSignal counter = new NumberSignal();

    private final NumberSignal sharedValue = new NumberSignal(0.5);

    public NumberSignal counter() {
        return counter;
    }

    public NumberSignal sharedValue() {
        return sharedValue;
    }
}
